### ISSUE TITLE
can-slot issues

### ISSUE THREAD

**Issue by roemhildtg** [canjs/can-stache#418]
__How often can you reproduce it?__ 

- [x] Always
- [ ] Sometimes
- [ ] Rarely
- [ ] Unable
- [ ] I didn’t try

__Description:__

There are some issues with `can-slot` and how the components are used with closing tags/empty tags.

https://jsbin.com/vavebuposo/1/edit?html,js,console,output

**Scenario 1:**
I don't want to override the slot, so I just use the shorthand component tag. 

`<component-name />` 

Workaround:  `<component-name></component-name>`

**Scenario 2:**
I want to override the default slot with an empty thing.

```
<component-name><can-template name="slotname"></can-template></component-name>
``` 

Workaround: 
```
<component-name><can-template name="slotname"><i /></can-template></component-name>
``` 

__Environment:__

| Software                   | Version
| -------------------- | -------
| can-stache version  | 3.14.2
| Browser                    |  Chrome
| Operating system    |  Windows 7

**Comment by justinbmeyer** [comment]
Thanks!

**Comment by chasenlehara** [comment]
Thanks @roemhildtg for the detailed issue and example! I’ve opened two PRs to fix these issues; should have them released next Tuesday.

**Comment by chasenlehara** [comment]
These issues should be fixed in [can-component 3.3.9](https://github.com/canjs/can-component/releases/tag/v3.3.9) and [can-stache 3.14.7](https://github.com/canjs/can-stache/releases/tag/v3.14.7)
